crates/hyperqueue/src/dashboard/state.rs

```diff
@@ -1,28 +1,51 @@
 use crate::dashboard::data::DashboardData;
+use crate::dashboard::ui::screen::controller::{DashboardScreenState, ScreenController};
 use crate::dashboard::ui::screen::Screen;
-use crate::dashboard::ui::screens::home::screen::HomeScreen;
+use crate::dashboard::ui::screens::home::screen::ClusterOverviewScreen;
+use crate::dashboard::ui::screens::worker::screen::WorkerOverviewScreen;
 use tako::common::WrappedRcRefCell;
 
 pub struct DashboardState {
-    screens: Vec<Box<dyn Screen>>,
-    current_screen: usize, // TODO: usize newtype wrapper or enum
+    cluster_overview_screen: ClusterOverviewScreen,
+    worker_overview_screen: WorkerOverviewScreen,
+
     data_source: WrappedRcRefCell<DashboardData>,
+
+    current_screen: DashboardScreenState,
+    controller: ScreenController,
 }
 
 impl DashboardState {
-    pub fn new(data_source: DashboardData) -> Self {
+    pub fn new(data_source: DashboardData, controller: ScreenController) -> Self {
         Self {
             data_source: WrappedRcRefCell::wrap(data_source),
-            screens: vec![Box::new(HomeScreen::default())],
-            current_screen: 0,
+            cluster_overview_screen: ClusterOverviewScreen::default(),
+            worker_overview_screen: WorkerOverviewScreen::default(),
+            current_screen: DashboardScreenState::ClusterOverviewScreen,
+            controller,
         }
     }
 
     pub fn get_data_source(&self) -> &WrappedRcRefCell<DashboardData> {
         &self.data_source
     }
 
-    pub fn get_current_screen_mut(&mut self) -> &mut dyn Screen {
-        self.screens[self.current_screen].as_mut()
+    /// Updates data for the next screen if required and changes the dashboard state to show it.
+    pub fn change_current_screen(&mut self, next_screen: DashboardScreenState) {
+        if let DashboardScreenState::WorkerOverviewScreen(worker_id) = next_screen {
```

Comment:
Please use a `match` here (without the `_` pattern), so that if new commands to change screen are added, we will have to add support for them here explicitly.